echopop/tests/test_operations.py

```diff
@@ -0,0 +1,450 @@
+            'station': np.concatenate( [ np.repeat( 'clouds' , 10 ) ,
+                                         np.repeat( 'waves' , 6 ) ] ) ,
+            'length': [ 4.0 , 5.0 , 5.0 , 4.0 , 5.0 , 5.0 , 4.0 , 6.0 , 
+                        6.0 , 7.0 , 2.0 , 4.0 , 3.0 , 2.0 , 4.0 , 3.0 ] ,
+            'length_bin': pd.cut( [ 4.0 , 5.0 , 5.0 , 4.0 , 5.0 , 5.0 , 4.0 , 6.0 , 
+                                    6.0 , 7.0 , 2.0 , 4.0 , 3.0 , 2.0 , 4.0 , 3.0 ] ,
+                                 np.array( [ 1.0 , 3.0 , 5.0 , 7.0 , 9.0 ] ) ) ,
+            'length_count': [ 1 , 1 , 2 , 1 , 1 , 2 , 1 , 1 , 1 , 1 ,
+                              10 , 20 , 30 , 30 , 20 , 10 ] ,
+        } ,
+    )
+
+    #----------------------------------
+    ### Run tests: `count_variable`
+    #----------------------------------
+    ### Check shape
+    assert eval_dataframe_monkey.shape == expected_dimensions
+    assert eval_dataframe_function.shape == expected_dimensions
+    ### Check output
+    assert np.all( eval_dataframe_monkey == expected_output )
+    assert np.all( eval_dataframe_function == expected_output )
+
+def test_stretch( ):
+
+    ### Create mock dataframe
+    test_dataframe = pd.DataFrame(
+        {
+            'stratum_num': [ 1 , 1 , 2 , 2 ] ,
+            'transect_num': [ 1 , 2 , 3 , 4 ] ,
+            'latitude': [ 0.0 , 1.0 , 3.0 , 4.0 ] ,
+            'longitude': [ -1.0 , 0.0 , 1.0 , 2.0 ] ,
+            'load_a_male': [ 5.0 , 4.0 , 2.0 , 1.0 ] ,
+            'load_a_female': [ 10.0 , 3.0 , 5.0 , 6.0 ] ,
+        } ,
+    )
+
+    ### Eval for later comparison 
+    # ---- Monkey patch method (TEMPORARY)
+    eval_dataframe_monkey = test_dataframe.stretch( variable = 'load_a' )
+    # ---- Normal function
+    eval_dataframe_function = stretch( test_dataframe , variable = 'load_a' )
+
+    ###--------------------------------
+    ### Expected outcomes
+    ###--------------------------------
+    # ---- Expected dimensions
+    expected_dimensions = tuple( [ 8 , 6 ] )
+    # ---- Expected output
+    expected_output = pd.DataFrame(
+        {
+            'transect_num': np.repeat( [ 1 , 2 , 3 , 4 ] , 2 ) ,
+            'latitude': np.repeat( [ 0.0 , 1.0 , 3.0 , 4.0 ] , 2 ) ,
+            'longitude': np.repeat( [ -1.0 , 0.0 , 1.0 , 2.0 ] , 2 ) ,
+            'stratum_num': np.repeat( [ 1 , 2 ] , 4 ) ,
+            'sex': np.tile( [ 'male' , 'female' ] , 4 ) ,
+            'load_a': [ 5.0 , 10.0 , 4.0 , 3.0 ,
+                        2.0 , 5.0 , 1.0 , 6.0 ] ,
+        } ,
+    )
+
+    #----------------------------------
+    ### Run tests: `count_variable`
+    #----------------------------------
+    ### Check shape
+    assert eval_dataframe_monkey.shape == expected_dimensions
+    assert eval_dataframe_function.shape == expected_dimensions
+    ### Check output
+    assert np.all( eval_dataframe_monkey == expected_output )
+    assert np.all( eval_dataframe_function == expected_output )
+
+def test_group_merge( ):
```

Comment:
This comment does not really belong to this PR, but since this function shows up here...

It seems that `group_merge` is only used once, and the content is very specific. I think in this case the operations can be exposed in the calling function `index_transect_age_sex_proportions` so that the code is easy to follow.